providers/aws/resources/aws.lr

```diff
@@ -1822,17 +1822,19 @@ private aws.s3.bucket.corsrule @defaults("name") {
 }
 
 // Amazon S3 bucket policy
-private aws.s3.bucket.policy @defaults("name version") {
+private aws.s3.bucket.policy @defaults("bucketName exists version") {
   // Unique ID for the policy
   id string
-  // Name for the policy
-  name string
+  // Bucket name that this policy belongs
+  bucketName string
   // Document for the policy
   document string
   // Version of the policy
   version() string
   // List of statements for the policy
   statements() []dict
+  // Whether the bucket policy exists
+  exists bool
```

Comment:
Buckets can have a policy, but not all of them has it. This is very helpful, I think.

Reply:
I don't understand why this is needed. If there is no bucket policy then that resource shouldn't be created, right? Do we have some specific requirement to have the MQL object even when such resource in AWS doesn't exist?

Reply:
we should be able to use null comparisons if we need to check that a policy exists or does not exist on a bucket. like aws.s3.bucket.policy != null -- does that not work?

Reply:
It sounds it is possible! 🙌🏽 Thank you both for the correction 💯